crates/indexer/src/lib.rs

```diff
@@ -138,12 +140,15 @@ where
                         continue;
                     }
 
-                    tracing::info!("Processing blocks from {} to {}", from_block, to_block);
+                    // cap to at most 500 blocks per batch
+                    let batch_end = min(to_block, from_block.saturating_add(MAX_BATCH_SIZE));
+
+                    tracing::info!("Processing blocks from {} to {}", from_block, batch_end);
 
-                    match self.process_blocks(from_block, to_block).await {
+                    match self.process_blocks(from_block, batch_end).await {
```

Comment:
This seems like it doesn't change the range at which the checkpoint is updated in `update_last_processed_block`, so this would just miss blocks past the batch size? Shouldn't this logic exist in `process_blocks`?